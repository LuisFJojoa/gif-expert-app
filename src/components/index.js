export * from "./AddCategory/AddCategory";
export * from "./GifExpertApp/GifExpertApp";
export * from "./GifGrid/GifGrid";
export * from "./GifItem/GifItem";
