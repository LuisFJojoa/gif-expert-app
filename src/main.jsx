import React from "react";
import ReactDOM from "react-dom/client";
import { GifExpertApp } from "./components";
import "./styles.css";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <GifExpertApp />
  </React.StrictMode>
);
