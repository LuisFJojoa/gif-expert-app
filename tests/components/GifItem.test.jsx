import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/";

describe("Pruebas en </GifItem>", () => {
  const title = "Kakarotto";
  const url = "https://kakaroto.com/goku.jpg";
  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);
    // expect(screen.getByRole("img").src).toBe(url);
    // expect(screen.getByRole("img").alt).toBe(title);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("Debe de mostrar el titulo en el componente", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
