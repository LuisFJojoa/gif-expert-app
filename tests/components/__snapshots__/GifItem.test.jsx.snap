// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en </GifItem> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Kakarotto"
      src="https://kakaroto.com/goku.jpg"
    />
    <p>
      Kakarotto
    </p>
  </div>
</div>
`;
